### PR TITLE
Apply limit interface and traits to `Number` rule

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #382

### CODE REVIEW

**Review by vjik — COMMENTED** [review]
I think not need use limit interface in `Number` rule.

1) More difficult trait: added float values + added `allowNegativeLimits` and `requireLimits` options.

2) In `Number` added overhead option `exactly`.

File: src/Rule/Trait/LimitTrait.php [hunk 1]
```diff
@@ -89,21 +89,23 @@ trait LimitTrait
...
-        int|null $max,
-        int|null $exactly,
+        int|float|null $min,
+        int|float|null $max,
+        int|float|null $exactly,
         string $lessThanMinMessage,
         string $greaterThanMinMessage,
-        string $notExactlyMessage
+        string $notExactlyMessage,
+        bool $requireLimits = true,
```
Comment (arogachev) [inline src/Rule/Trait/LimitTrait.php:107]: Need to add PHPDoc with description.

**Comment by samdark** [comment]
$equal = \abs($a - $b) < PHP_FLOAT_EPSILON;

Should be used for floats.

**Comment by vjik** [comment]
A little discussed. Idea: make improvements of `Number` (names, floats), but don't change  `LimitTrait` and don't use it into `Number`.

**Pull request closed without merging**